taichi/ir/frontend_ir.h

```diff
@@ -392,9 +392,13 @@ class ExternalTensorExpression : public Expression {
   DataType dt;
   int dim;
   int arg_id;
+  int element_dim;
```

Comment:
nit1: Add comment saying that this encodes SOA/SOA
nit2: init these members

Reply:
nit1: done
nit2: I can't think of any meaningful initial values for these members.. Other expressions don't init them either.